python/tchannel/tornado/connection.py

```diff
@@ -456,3 +450,87 @@ def ping(self):
 
     def pong(self):
         return self._write(messages.PingResponseMessage())
+
+
+class StreamConnection(TornadoConnection):
+    def __init__(self, connection):
+        super(StreamConnection, self).__init__(connection)
+
+    @tornado.gen.coroutine
+    def _stream(self, context, message_factory):
+        """write request/response into frames
+
+        Transform request/response into protocol level message objects based on
+        types and argstreams.
+
+        Assumption: the chunk data read from stream can fit into memory.
+
+        If arg stream is at init or streaming state, build the message based on
+        current chunk. If arg stream is at completed state, put current chunk
+        into args[] array, and continue to read next arg stream in order to
+        compose a larger message instead of sending multi small messages.
+
+        Note: the message built at this stage is not guaranteed the size is
+        less then 64KB.
+
+        Possible messages created sequence:
+
+        Take request as an example:
+        CallRequestMessage(flags=fragment)
```

Comment:
Use proper reStructuredText for this. Add a colon to "example:" to indicate that the paragraph that follows is a code block, add an empty line, and indent the example messages 4 spaces. Like so,

    Take request as an example::

        CallRequestMessage(flags=fragment)
        [..]